test/setup_environment.sh

```diff
@@ -162,6 +162,38 @@ pipenv run make -j$(nproc) kkemu
 find . -name "emulator.img" -exec rm {} \;
 cd ..
 
+# Clone ledger simulator Speculos if it doesn't exist, or update it if it does
+speculos_setup_needed=false
+if [ ! -d "speculos" ]; then
+    git clone --recursive https://github.com/LedgerHQ/speculos.git
```

Comment:
we should choose a specific hash to target a commit

Reply:
We don't do that for the other simulators...